extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -663,4 +515,184 @@ export abstract class AzureAuth implements vscode.Disposable {
 
 		return account;
 	}
+
+	//#endregion
+
+	//#region network functions
+	protected async makePostRequest(url: string, postData: AuthorizationCodePostData | TokenPostData | DeviceCodeStartPostData | DeviceCodeCheckPostData): Promise<AxiosResponse<any>> {
+		const config: AxiosRequestConfig = {
+			headers: {
+				'Content-Type': 'application/x-www-form-urlencoded'
+			},
+			validateStatus: () => true // Never throw
+		};
+
+		// Intercept response and print out the response for future debugging
+		const response = await axios.post(url, qs.stringify(postData), config);
+		Logger.pii(url, postData, response.data);
+		return response;
+	}
+
+	private async makeGetRequest(url: string, token: string): Promise<AxiosResponse<any>> {
+		const config: AxiosRequestConfig = {
+			headers: {
+				'Content-Type': 'application/json',
+				'Authorization': `Bearer ${token}`
+			},
+			validateStatus: () => true // Never throw
+		};
+
+		const response = await axios.get(url, config);
+		Logger.pii(url, response.data);
+		return response;
+	}
+
+	//#endregion
+
+	//#region inconsequential
+	protected getTokenClaims(accessToken: string): TokenClaims | undefined {
+		try {
+			const split = accessToken.split('.');
+			return JSON.parse(Buffer.from(split[1], 'base64').toString('binary'));
+		} catch (ex) {
+			throw new Error('Unable to read token claims: ' + JSON.stringify(ex));
+		}
+	}
+
+	protected toBase64UrlEncoding(base64string: string): string {
+		return base64string.replace(/=/g, '').replace(/\+/g, '-').replace(/\//g, '_'); // Need to use base64url encoding
+	}
+
+	public async deleteAllCache(): Promise<void> {
+		const results = await this.tokenCache.findCredentials('');
```

Comment:
Definitely add a test for this and/or findCredentials to validate this logic stays consistent. It's something I could easily see somebody changing without realizing some places depend on it handling empty strings like this.